### PR TITLE
Detail service implementations and add new APIs for the authentication service

### PR DESCRIPTION
**Pull request by mssun**
## Description

This PR introduce some changes on the services:
 - Support Ubuntu 20.04 docker image
 - Support intermediate cert in DCAP attestation cert
 - Fix RPC error handling
 - Informative logs
 - Fix reuse port issue
 - Port probing for service launching
 - New APIs for authentication service

## Type of change (select or add applied and delete the others)

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] API change with a documentation update
- [ ] Additional test coverage
- [ ] Code cleanup or just sync with upstream third-party crates

## How has this been tested?

CI test: https://ci.mesalock-linux.org/mssun/incubator-teaclave/112

## Checklist

- [x] Fork the repo and create your branch from `master`.
- [x] If you've added code that should be tested, add tests.
- [x] If you've changed APIs, update the documentation.
- [x] Ensure the tests pass (see CI results).
- [x] Make sure your code lints/format.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -167,32 +316,32 @@ name: sgx-release-ubuntu-1804
...
 - name: check
-  image: teaclave/teaclave-build-ubuntu-1804-sgx-2.14:0.1.2
+  image: teaclave/teaclave-build-ubuntu-1804-sgx-2.14:0.1.3
   commands:
   - . /root/.cargo/env
   - . /opt/sgxsdk/environment
   - cd build && make check
 - name: compile
-  image: teaclave/teaclave-build-ubuntu-1804-sgx-2.14:0.1.2
+  image: teaclave/teaclave-build-ubuntu-1804-sgx-2.14:0.1.3
```
Comment (bigdata-memory) [inline .drone.yml:332]: The repeated image values can be optimized by `Anchor`, it is a Yaml feature, so we can safely apply it to any Yaml files.

Comment (mssun) [reply .drone.yml:332]: Thanks, this is very helpful!

File: rpc/src/transport.rs [hunk 2]
```diff
@@ -96,19 +96,19 @@ where
...
                 Err(e) => match e {
-                    protocol::ProtocolError::IoError(_) => {
-                        debug!("Connection disconnected.");
+                    protocol::ProtocolError::IoError(e) => {
+                        log::debug!("Connection disconnected: {:?}", e);
                         return Ok(());
                     }
                     _ => {
-                        debug!("{:?}", e);
+                        warn!("Connection error: {:?}", e);
```
Comment (ya0guang) [inline rpc/src/transport.rs:104]: `log::warn!` or `warn!`?

File: sdk/python/teaclave.py [hunk 3]
```diff
@@ -237,10 +420,95 @@ def user_login(self, user_id: str, user_password: str) -> str:
...
+        """
+        self.check_channel()
+        self.check_metadata()
+        request = ResetUserPasswordRequest(self.metadata, user_id)
+        _write_message(self.channel, request)
+        response = _read_message(self.channel)
+        if response["result"] == "ok":
+            return response["content"]["token"]
+        else:
+            reason = "unknown"
```
Comment (ya0guang) [inline sdk/python/teaclave.py:464]: adjust the `"unknown"`s in the future

**Review by ya0guang — COMMENTED** [review]
Looks good to me, some minor places can be fixed in the future.

**Review by ya0guang — APPROVED** [review]
LGTM!

**Pull request merged**